src/main/java/com/hw/bookstore/service/BookServiceImpl.java

```diff
@@ -0,0 +1,38 @@
+package com.hw.bookstore.service;
+
+import com.hw.bookstore.domain.entity.Book;
+import com.hw.bookstore.domain.repository.BookRepository;
+import com.hw.bookstore.dto.BookDto;
+import com.hw.bookstore.dto.CreateBookRequestDto;
+import com.hw.bookstore.mapper.BookMapper;
+import java.util.List;
+import lombok.RequiredArgsConstructor;
+import org.springframework.stereotype.Service;
+
+@Service
+@RequiredArgsConstructor
+public class BookServiceImpl implements BookService {
+
+    private final BookRepository bookRepository;
+    private final BookMapper bookMapper;
+
+    @Override
+    public BookDto save(CreateBookRequestDto requestDto) {
+        Book book = bookMapper.toBook(requestDto);
+        bookRepository.save(book);
+        return bookMapper.toBookDto(book);
+    }
+
+    @Override
+    public List<BookDto> findAll() {
+        return bookRepository.findAll().stream()
+                .map(bookMapper::toBookDto)
+                .toList();
+    }
+
+    @Override
+    public BookDto findById(Long id) {
+        Book book = bookRepository.findById(id);

```

Comment:
you should return Optional from the repository level and throw EntityNotFoundException here

Reply:
I catch EntityNotFoundExceptionin in the BookRepositoryImpl class ,  so Optional is redundant in this case.